src/XMLexport.cpp

```diff
@@ -39,619 +40,795 @@ using namespace std;
-        {
-            ret = writeAlias( pChildAlias );
-        }
-    }
-    writeEndElement();
-
-    writeStartElement("ActionPackage");
-    for(auto it4 = pT->mActionUnit.mActionRootNodeList.begin(); it4 != pT->mActionUnit.mActionRootNodeList.end(); it4++)
-    {
-        TAction * pChildAction = *it4;
-        if (pChildAction->mModuleMember) continue;
-        ret = writeAction( pChildAction );
-    }
-    writeEndElement();
-
-    writeStartElement("ScriptPackage");
-    for(auto it5 = pT->mScriptUnit.mScriptRootNodeList.begin(); it5 != pT->mScriptUnit.mScriptRootNodeList.end(); it5++)
-    {
-        TScript * pChildScript = *it5;
-        if (pChildScript->mModuleMember) continue;
-        ret = writeScript( pChildScript );
-    }
-    writeEndElement();
-
-    writeStartElement("KeyPackage");
-    for(auto it6 = pT->mKeyUnit.mKeyRootNodeList.begin(); it6 != pT->mKeyUnit.mKeyRootNodeList.end(); it6++)
-    {
-        TKey * pChildKey = *it6;
-        if (pChildKey->mModuleMember) continue;
-        ret = writeKey( pChildKey );
-    }
-    writeEndElement();
-
-    writeStartElement("VariablePackage");
-    LuaInterface * lI = pT->getLuaInterface();
-    VarUnit * vu = lI->getVarUnit();
-    //do hidden variables first
-    writeStartElement("HiddenVariables");
-    QSetIterator<QString> it8( vu->hiddenByUser );
-    while( it8.hasNext() )
-    {
-        writeTextElement( "name", it8.next() );
-    }
-    writeEndElement();
-    TVar * base = vu->getBase();
-    if ( !base )
-    {
-        lI->getVars( false );
-        base = vu->getBase();
-    }
-    if ( base )
-    {
-        QListIterator<TVar *> it7( base->getChildren(false) );
-        while( it7.hasNext() )
-        {
-            TVar * var = it7.next();
-            writeVariable( var, lI, vu );
-        }
-    }
-    writeEndElement();
-
-    return ret;
+        writeStartElement( "mInstalledPackages" );
+        for( int i = 0; i < pHost->mInstalledPackages.size(); ++i ) {
+            writeTextElement( "string", pHost->mInstalledPackages.at( i ) );
+        }
+        writeEndElement(); // </mInstalledPackages>
+
+        if( pHost->mInstalledModules.size() ) {
+            writeStartElement( "mInstalledModules" );
+            QMapIterator<QString, QStringList> it(pHost->mInstalledModules);
+            pHost->modulesToWrite.clear();
+            while( it.hasNext() ) {
+                it.next();
+                writeTextElement("key", it.key());
+                QStringList entry = it.value();
+                writeTextElement("filepath", entry.at(0));
+                writeTextElement("globalSave", entry.at(1));
+                if( entry.at(1).toInt() ) {
+                    pHost->modulesToWrite.insert( it.key(), entry );
+                }
+                writeTextElement("priority", QString::number(pHost->mModulePriorities.value(it.key())));
+            }
+            writeEndElement(); // </mInstalledModules>
+        }
+// CHECK: Do we need:
```

Comment:
I guess not as our reader code gets along quite nicely without a `<mInstalledModules>`...`</mInstalledModules>` segment...